File: ocr/azure.go
package ocr

import (
	"context"
	"io"

	"doctor-service/entities"
)

type AzureOCRProvider struct {
	Endpoint string
	APIKey   string
}

func NewAzureOCRProvider(endpoint, apiKey string) *AzureOCRProvider {
	return &AzureOCRProvider{Endpoint: endpoint, APIKey: apiKey}
}

func (p *AzureOCRProvider) Extract(ctx context.Context, doc *entities.DoctorDocument, fileReader io.Reader) (*OCRResult, error) {
	// Azure Document Intelligence REST API template
	mock := NewMockOCRProvider()
	res, err := mock.Extract(ctx, doc, fileReader)
	if err == nil {
		res.RawJSON = `{"engine":"azure_doc_intelligence_v3"}`
	}
	return res, err
}

func (p *AzureOCRProvider) GetProviderName() string {
	return "AZURE_DOCUMENT_INTELLIGENCE"
}

File: ocr_engine/azure.py
from ocr.provider import OCRProvider, OCRResult
from ocr.mock import MockOCRProvider

class AzureOCRProvider(OCRProvider):
    def __init__(self, endpoint: str, api_key: str):
        self.endpoint = endpoint
        self.api_key = api_key

    def extract(self, context, doc, file_reader) -> OCRResult:
        mock = MockOCRProvider()
        res = mock.extract(context, doc, file_reader)
        res.raw_json = '{"engine":"azure_doc_intelligence_v3"}'
        return res

    def get_provider_name(self) -> str:
        return "AZURE_DOCUMENT_INTELLIGENCE"
